javascript/ql/experimental/adaptivethreatmodeling/lib/experimental/adaptivethreatmodeling/EndpointFeatures.qll

```diff
@@ -145,6 +159,65 @@ module FunctionBodies {
   }
 }
 
+/** This module provides functionality for getting the local neighborhood around an AST node within its enclosing function body, providing a locally-scoped version of the `enclosingFunctionBody` feature. */
+module NeighborhoodBodies {
+  /**
+   * Return the ancestor of the input AST node that has the largest number of descendants (i.e. the node nearest the
+   * root) but has no more than 128 descendants.
+   * TODO: Maybe instead of a threshold on number of descendants, we should instead have a threshold on the number of
+   * leaves in the subtree, which is a closer approximation to the number of tokens in the subtree.
+   */
+  Raw::AstNode getNeighborhoodAstNode(Raw::AstNode node) {
+    if getNumDescendents(node.getParentNode()) > 128
```

Comment:
What happens if the root of the AST has <128 descendants? What does `node.getParentNode()` do if `node` has no parent? Maybe we need to explicitly check whether `node` has a parent, and if not do `result = node`?

Reply:
Great shout, done in 0dc2cef05f144c20578437bac9b9c0f0bc11d93c